draft-ietf-rtcweb-jsep.xml

```diff
@@ -1626,36 +1626,53 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
-          generated MID (see
-          <xref target="sec.applying-a-remote-desc" />), which is used
-          in place of a supplied MID. If these values or the candidate
-          string are invalid, an error is generated.</t>
-
-          <t>The purpose of the ufrag is to resolve ambiguities when
-          trickle ICE is in progress during an ICE restart. If the
-          ufrag is absent, the candidate MUST be assumed to belong to
-          the most recently applied remote description. Connectivity
-          checks will be sent to the new candidate.</t>
-
-          <t>This method can also be used to provide an
-          end-of-candidates indication to the ICE agent, as defined in
-          <xref target="I-D.ietf-ice-trickle" />). The MID and ufrag
-          are used as described above to determine the m= section and
-          ICE generation for which candidate gathering is complete. If
-          the ufrag is not present, then the end-of-candidates
-          indication MUST be assumed to apply to the relevant m=
-          section in the most recently applied remote description. If
-          neither the MID nor the m= index is present, then the
+          the rules defined for Trickle ICE. Otherwise, the IceCandidate
+          is treated as an end-of-candidates indication, as defined in
+          <xref target="I-D.ietf-ice-trickle" />).</t>
+          
+          <t>In either case, the MID and ufrag fields from the supplied 
+          IceCandidate are used to
+          determine which m= section and ICE candidate generation the
+          IceCandidate is for. In particular, the ufrag is used to resolve
+          ambiguities when trickle ICE is in progress during an ICE restart.</t>
+          
+          <t>Specific processing for each case is outlined below. If any
+          IceCandidate fields contain invalid values, or an error
+          occurs during the processing, the supplied IceCandidate MUST be ignored
+          and an error MUST be returned.</t>
+      
+          <t>If the IceCandidate is a new remote candidate:
+            <list style="symbols">
+              <t>The m= section index field must be present.</t>
+              
+              <t>If the MID field is not filled in, the m=
+              section index MUST be used to look up the locally generated
+              MID (see <xref target="sec.applying-a-remote-desc" />), which
+              is used in place of a supplied MID.</t>
+          
+              <t>If the ufrag field is not filled in, the candidate object MUST be
+              assumed to refer to the most recently applied remote description.</t>
+              
+              <t>If the candidate is valid, it is added to the ICE Agent as a new
+              remote candidate. Connectivity checks will be sent to the new candidate.</t>
+            </list>
+          </t>
+          
+          <t>If the IceCandidate is an end-of-candidates indication:
```

Comment:
again here, I would tend to say "candidate field is empty" instead of end-of-candiates